File: baselib/src/main/java/com/stjy/baselib/net/net1/callback/BaseVODCallback.java
package com.stjy.baselib.net.net1.callback;


import com.alibaba.sdk.android.vod.upload.model.UploadFileInfo;

/**
 * @author daifalin
 * @date 2019/3/29 2:45 PM
 * @ClassName BaseVODCallback
 * @Description
 */
public abstract class BaseVODCallback {
    public void onUploadSucceed(UploadFileInfo info) {

    }

    public abstract void onUploadFinished(String videoUrl, String imageUrl);

    public abstract void onUploadFailed(UploadFileInfo info, String code, String message);

    public void onUploadProgress(UploadFileInfo info, long uploadedSize, long totalSize) {

    }

    public void onUploadStarted(UploadFileInfo uploadFileInfo) {

    }

}

File: baselib/src/main/java/com/stjy/baselib/net/net1/callback/BaseVODCallback.kt
package com.stjy.baselib.net.net1.callback

import com.alibaba.sdk.android.vod.upload.model.UploadFileInfo

/**
 * @author daifalin
 * @date 2019/3/29 2:45 PM
 * @ClassName BaseVODCallback
 * @Description
 */
abstract class BaseVODCallback {
    fun onUploadSucceed(info: UploadFileInfo?) {}
    abstract fun onUploadFinished(videoUrl: String?, imageUrl: String?)
    abstract fun onUploadFailed(info: UploadFileInfo?, code: String?, message: String?)
    fun onUploadProgress(info: UploadFileInfo?, uploadedSize: Long, totalSize: Long) {}
    fun onUploadStarted(uploadFileInfo: UploadFileInfo?) {}
}